homeassistant/components/light/__init__.py

```diff
@@ -282,7 +282,14 @@ class SetIntentHandler(intent.IntentHandler):
                 pars = params.copy()
                 pars[ATTR_PROFILE] = Profiles.get_default(light.entity_id)
                 preprocess_turn_on_alternatives(pars)
-            await light.async_turn_on(**pars)
+            if ATTR_BRIGHTNESS in pars and pars[ATTR_BRIGHTNESS] == 0:
```

Comment:
Don't do this inside the loop. All parameters should be processed before we start looping over lights.

Reply:
The parameters are written to in the loop though.

Reply:
`params` is not.

Reply:
Alright, I shuffled it around a bit so `params` is only updated once outside the loop.